src/main/utility/mod.rs

```diff
@@ -84,6 +84,7 @@ impl<T> HostTreePointer<T> {
     /// SAFETY: Pointer must only be dereferenced while the configures Host is
     /// still active, in addition to the normal safety requirements for
     /// dereferencing a pointer.
+    #[allow(unused_must_use)]
```

Comment:
This should be left as a warning. It's being tracked at https://github.com/shadow/shadow/issues/2514.